jerry-core/ecma/builtin-objects/ecma-builtin-math.c

```diff
@@ -570,38 +570,32 @@ ecma_builtin_math_dispatch_routine (uint16_t builtin_routine_id, /**< built-in w
       }
       case ECMA_MATH_OBJECT_EXPM1:
       {
-#ifdef JERRY_LIBM_MATH_H
-        return ecma_raise_type_error (ECMA_ERR_MSG ("UNIMPLEMENTED: Math.expm1"));
-#else /* !JERRY_LIBM_MATH_H */
-        x = DOUBLE_TO_ECMA_NUMBER_T (expm1 (x));
-#endif /* JERRY_LIBM_MATH_H */
+        /* ES6 20.2.2.15 Math.expm1(-0.0) = -0.0 */
+        if ((x == 0.0) && (ecma_number_is_negative (x)))
+        {
+          break;
+        }
```

Comment:
This check should be inside #ifdef JERRY_LIBM_MATH_H guard,
because exp in libm conforms to the spec without this workaround.

Reply:
exp(x) function conforms to specification, but for (0-) "left" and (0+) "right" "0" - but in this case it should distinguish "left" and "right" "1", because we are subtracting 1 from exp(x) result.

I deleted this check, build with _tools/build.py --profile=es2015-subset --debug --show-opcodes=ON --jerry-libm=OFF --link-lib=m --clean_ and unfortunately test fails.

Reply:
Ah, I got it. In this case we should use expm1 and log1p instead of exp and log if we build with (GNU) libm, the new tests pass with them.

Reply:
I will add native implementation (not reuse of exp()) to provide better accuracy for small numbers (both for expm1 and log1p).